internal/decomposer/decomposer.go

```diff
@@ -65,6 +68,13 @@ func (decomposer *Decomposer) Execute(playbook cacao.Playbook) (*ExecutionDetail
 	variables := cacao.NewVariables()
 	variables.Merge(playbook.PlaybookVariables)
 
+	// Reporting workflow instantiation
+	metadata := execution.Metadata{
```

Comment:
You're creating a metadata object here and only using the ExecutionId just pass that to ReportWorkflow